### PR TITLE
BXMSPROD-944 GA build-chain pull-request flow

### PR DESCRIPTION
**Pull request by Ginxo**
https://issues.redhat.com/browse/BXMSPROD-944

related PRs:
* https://github.com/kiegroup/kogito-apps/pull/535
* https://github.com/kiegroup/kogito-examples/pull/452
* https://github.com/kiegroup/kogito-runtimes/pull/887
* https://github.com/kiegroup/kogito-pipelines/pull/158

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          wget -P $M2_HOME/lib/ext https://repo1.maven.org/maven2/io/takari/takari-filemanager/0.8.3/takari-filemanager-0.8.3.jar
+          wget -P $M2_HOME/lib/ext https://repo1.maven.org/maven2/io/takari/maven/takari-smart-builder/0.6.1/takari-smart-builder-0.6.1.jar
+      - name: Install takari (windows)
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: |
+          (New-Object System.Net.WebClient).DownloadFile("https://repo1.maven.org/maven2/io/takari/aether/takari-local-repository/0.11.3/takari-local-repository-0.11.3.jar", "$Env:M2_HOME\lib\ext\takari-local-repository-0.11.3.jar")
+          (New-Object System.Net.WebClient).DownloadFile("https://repo1.maven.org/maven2/io/takari/takari-filemanager/0.8.3/takari-filemanager-0.8.3.jar", "$Env:M2_HOME\lib\ext\takari-filemanager-0.8.3.jar")
+          (New-Object System.Net.WebClient).DownloadFile("https://repo1.maven.org/maven2/io/takari/maven/takari-smart-builder/0.6.1/takari-smart-builder-0.6.1.jar", "$Env:M2_HOME\lib\ext\takari-smart-builder-0.6.1.jar")
+      # See https://docs.github.com/en/actions/guides/building-and-testing-java-with-maven#caching-dependencies
```
Comment (mareknovotny) [inline .github/workflows/pull-request.yml]: maybe we don't need this extension as here we don't use parallel builds at all

Comment (Ginxo) [reply .github/workflows/pull-request.yml]: removed

**Review by mareknovotny — APPROVED** [review]
lgtm

File: .ci/project-dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  - project: kiegroup/kogito-runtimes
+
+  - project: kiegroup/kogito-apps
+    dependencies:
+      - project: kiegroup/kogito-runtimes
+
+  - project: kiegroup/kogito-examples
+    dependencies:
+      - project: kiegroup/kogito-apps
+      - project: kiegroup/kogito-runtimes
```
Comment (danielezonca) [inline .ci/project-dependencies.yaml]: Please remove this, examples is no more a downstream project of kogito-apps

File: .ci/pull-request-config.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  - project: kiegroup/kogito-examples
+    build-command:
+      current: |
+        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+        mvn -f kogito-examples-persistence/pom.xml clean install -Ppersistence -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+        mvn -f kogito-examples-events/pom.xml clean install -Pevents -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+    clone:
+      - kogito-examples-persistence
+      - kogito-examples-events
```
Comment (danielezonca) [inline .ci/pull-request-config.yaml]: To be removed

File: .ci/pull-request-config.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  - project: kiegroup/kogito-runtimes
+    build-command:
+      upstream: |
+        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+      current: |
+        mvn clean install -Prun-code-coverage -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+        mvn validate -Psonarcloud-analysis -e -nsu -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+        mvn -f kogito-runtimes-integration-tests/pom.xml clean
+    clone:
+      - kogito-runtimes-integration-tests
```
Comment (danielezonca) [inline .ci/pull-request-config.yaml]: To be removed

File: .ci/pull-request-config.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  - project: kiegroup/kogito-apps
+    build-command:
+      upstream: |
+        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
```
Comment (danielezonca) [inline .ci/pull-request-config.yaml]:
```suggestion
        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3 -DskipTests -DskipITs
```

File: .ci/pull-request-config.yaml (hunk 4 above)
Comment (Ginxo) [reply .ci/pull-request-config.yaml]: can you please confirm these are the lines to be removed? @danielezonca 
```
mvn -f kogito-runtimes-integration-tests/pom.xml clean
    clone:
      - kogito-runtimes-integration-tests
```

File: .ci/pull-request-config.yaml [hunk 6]
```diff
@@ -17,7 +17,7 @@ build:
   - project: kiegroup/kogito-apps
     build-command:
       upstream: |
-        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3 -DskipTests -DskipITs
```
Comment (mareknovotny) [inline .ci/pull-request-config.yaml]: Are tests disabled on purpose?

Comment (Ginxo) [reply .ci/pull-request-config.yaml]: @danielezonca ^ ¿?

Comment (mareknovotny) [reply .ci/pull-request-config.yaml]: of course they are, i didn'ŧ notice it is upstream command 🔢

**Review by danielezonca — APPROVED** [review]

**Pull request merged**